sigstore/sign.py

```diff
@@ -293,7 +308,7 @@ def with_signer(
         to sign different artifacts.
         Default is `True`.
         """
-        signer = Signer(identity_token, cache)
+        signer = Signer(identity_token, self._fulcio, cache)
```

Comment:
We can make this a little more resilient to API changes by pushing the whole `SigningContext` in, i.e.:

```suggestion
        signer = Signer(identity_token, self, cache)
```

(and then using `SigningContext._fulcio` wherever we need it.)